cardano-cli/src/Cardano/CLI/Shelley/Commands.hs

```diff
@@ -303,3 +312,21 @@ newtype TxFile
 newtype VerificationKeyBase64
   = VerificationKeyBase64 String
   deriving (Eq, Show)
+
+-- | Either a verification key, verification key hash, or path to a
+-- verification key file.
+data VerificationKeyOrHashOrFile keyrole
+  = VerificationKeyValue !(VerificationKey keyrole)
+  -- ^ A verification key.
+  | VerificationKeyHash !(Hash keyrole)
+  -- ^ A verification key hash.
+  | VerificationKeyFilePath !VerificationKeyFile
+  -- ^ A path to a verification key file.
+  -- Note that this file hasn't been validated at all (whether it exists,
+  -- contains a key of the correct type, etc.)
+
+deriving instance (Show (VerificationKey keyrole), Show (Hash keyrole))
+  => Show (VerificationKeyOrHashOrFile keyrole)
+
+deriving instance (Eq (VerificationKey keyrole), Eq (Hash keyrole))
+  => Eq (VerificationKeyOrHashOrFile keyrole)
```

Comment:
Is this the same as `deriving (Eq, Show)`?

Reply:
Yeah, but I couldn't do that directly since I needed the constraints: `Eq (VerificationKey keyrole), Eq (Hash keyrole)`.